JT/01_environment_setup/README.adoc

```diff
@@ -0,0 +1,406 @@
+
+To set main as the default branch name, do:
+
+[source,bash]
+----
+$ git config --global init.defaultBranch main
+----
+
+If you want to check your configuration settings, you can use the git config --list command to list all the settings Git can find at that point:
+
+[source,bash]
+----
+$ git config --list
+user.name=John Doe
+user.email=john.doe@endava.com
+color.status=auto
+color.branch=auto
+color.interactive=auto
+color.diff=auto
+...
+----
+You can also check what Git thinks a specific key's value is by typing git config <key>:
+
+[source,bash]
+----
+$ git config user.name
+John Doe
+----
+=== [[intellij-idea]]IntelliJ IDEA
+
+. Download and install IntelliJ IDEA
++
+Link for downloads: https://www.jetbrains.com/idea/download/
++
+image::resources/idea-01.png[]
+
+. Set up IntelliJ IDEA
++
+* Open IntelliJ IDEA.
+Upon the first launch, you'll have to accept the JetBrains Community Edition Terms and Privacy Policy. Click on "Accept" to proceed.
++
+image::resources/idea-02.png[]
+
+Then you can see the first screen of IntelliJ IDEA. Here you can:
+
+image::resources/idea-03.png[]
+
+
+. Install plugins for IntelliJ IDEA
++
+* Gherkin
+* Cucumber for Java
++
+image::resources/idea-04.png[]
++
+* Lombok
++
+image::resources/idea-06.png[]
+
+. Set up the new project in IntelliJ IDEA or create a new one from existing sources
++
+I. You should have a link to the project repository. If you don't have it, ask your team lead for it.
++
+II. OR:
+* Set a project name.
+* Choose where the project will be created.
+* Choose Language: Java.
+* Choose Build system: Maven.
+* Choose SDK: 21 (or the version you have installed).
+* Click Finish.
+
+image::resources/idea-07.png[]
+
+* Open the project structure and set the project SDK and language level (Press Ctrl+Alt+Shift+S).
++
+image::resources/idea-08.png[]
++
+image::resources/idea-09.png[]
+
+=== [[postman]]Postman
+
+Postman is an app for interacting with HTTP APIs. It presents you with a friendly GUI for constructing requests and reading responses.
+
+But Postman recently changed their software to remove the Scratch Pad mode, which means that all configuration data, including Collections and Environments, is uploaded to Postman's cloud servers. This poses a **potential security risk** for many projects. To avoid storing collections in a Postman account, users are advised to uninstall their current version of Postman and install an older version using the following link:
+
+link:https://go.pstmn.io/dl-win64-v9-latest[https://go.pstmn.io/dl-win64-v9-latest]
+
+. Download Postman for Windows (64-bit), from the following link: https://go.pstmn.io/dl-win64-v9-latest.
++
+image::resources/postman-01.png[]
+
+. Install Postman
++
+image::resources/postman-02.png[]
+
+**Do not sign up or log in and don't Update software for security reasons.**
```

Comment:
Please check if you it's valuable for you to use Adoc notification types like [WARNING] in text where it could be useful and more noticeable.

Reply:
Thank you for the good advice. Done